packages/@n8n/db/src/migrations/common/1760314000000-CreateWorkflowDependencyTable.ts

```diff
@@ -0,0 +1,32 @@
+import type { MigrationContext, ReversibleMigration } from '../migration-types';
+
+export class CreateWorkflowDependencyTable1760314000000 implements ReversibleMigration {
+	async up({ schemaBuilder: { createTable, column } }: MigrationContext) {
+		await createTable('workflow_dependency')
+			.withColumns(
+				column('id').uuid.primary,
+				column('workflowId').varchar(36).notNull,
+				column('workflowVersionId').int.notNull.comment('Version of the workflow'),
+				column('dependencyType')
+					.varchar(32)
+					.notNull.comment(
+						'Type of dependency: "credential", "nodeType", "webhookPath", or "workflowCall"',
+					),
+				column('dependencyId').varchar(255).notNull.comment('ID or name of the dependency'),
+				column('nodeId').varchar(255).comment('Node ID for nodeType dependencies, null otherwise'),
+				column('indexVersionId').int.notNull.default(1).comment('Version of the index structure'),
+			)
+			.withForeignKey('workflowId', {
+				tableName: 'workflow_entity',
+				columnName: 'id',
+				onDelete: 'CASCADE',
+			})
+			.withIndexOn(['workflowId'])
+			.withIndexOn(['dependencyType'])
+			.withIndexOn(['dependencyId']).withTimestamps;
```

Comment:
Since we're gonna drop and recreate the index for each workflow every time, the update timestamp will always be the same as `createdAt`. I think we could only include the `createdAt` timestamp. Not sure if the DSL allows adding only `createdAt`. We could add it there if not